core_types/src/dna/mod.rs

```diff
@@ -178,13 +178,19 @@ impl Dna {
     }
 
     /// Return the name of the zome holding a specified app entry_type
-    pub fn get_zome_name_for_entry_type(&self, entry_type_name: &str) -> Option<String> {
+    pub fn get_zome_name_for_app_entry_type(
+        &self,
+        app_entry_type: &AppEntryType,
+    ) -> Option<String> {
+        let entry_type_name = String::from(app_entry_type.to_owned());
         // pre-condition: must be a valid app entry_type name
-        assert!(EntryType::has_valid_app_name(entry_type_name));
+        assert!(EntryType::has_valid_app_name(&entry_type_name));
         // Browse through the zomes
         for (zome_name, zome) in &self.zomes {
             for (zome_entry_type_name, _) in &zome.entry_types {
-                if *zome_entry_type_name == entry_type_name {
+                if *zome_entry_type_name
+                    == EntryType::App(AppEntryType::from(entry_type_name.to_string()))
```

Comment:
yech, this seems so much clunkier that what it was. Sorry, just griping about rust, not really a useful comment, though I do see that his double nested for loop could be refactored into a closure and use by both `get_zome_name_for_app_entry_type` and `get_entry_type_def`  maybe a folowup?

Reply:
@zippy `EntryType::from(entry_type_name)` probably works too